backend/src/main/java/com/huellapositiva/infrastructure/orm/repository/JpaContactPersonRepository.java

```diff
@@ -35,4 +35,12 @@
 
     @Query("FROM JpaContactPerson cp LEFT JOIN FETCH cp.joinedEsal je WHERE je.id = :id")
     Optional<JpaContactPerson> findByEsalId(@Param("id") String id);
+
+    @Query("FROM JpaContactPerson cp LEFT JOIN FETCH cp.contactPersonProfile p LEFT JOIN FETCH cp.credential c WHERE c.id = :accountId")
+    Optional<JpaContactPerson> findByAccountIdWithProfile(@Param("accountId") String accountId);
+
+    @Modifying
+    @Transactional
+    @Query("UPDATE JpaContactPerson cp SET cp.contactPersonProfile = :profile WHERE cp.id = :id")
+    Integer updateProfile(@Param("id") String id, @Param("profile") JpaContactPersonProfile profile);
```

Comment:
This query should be removed.

In ORM world the relations are not persisted on that way. You need to have the two objects belonging to the relation loaded in the Hibernate session (first level cache).

That means that if you save the object that owns (contains foreign key in its DB table) the relation Hibernate will handle the rest for you.

```
JpaContactPerson jpaContactPerson = ...
JpaContactPersonProfile jpaContactPersonProfile = '... -> this object must be in the first level cache (loaded from DB or persisted reference)
jpaContactPerson.setProfile(jpaContactPersonProfile);
jpaContactPersonRepository.save(jpaContactPerson); -> Hibernate will see the relation and get the profile id to persist it in the JpaContactPerson table
```